src/main/kotlin/de/livepoll/api/service/PollService.kt

```diff
@@ -123,71 +158,123 @@ class PollService(
                 this.startDate = null
                 this.endDate = null
             }
-            if (poll.slug != null && isSlugUnique(poll.slug)) {
-                this.slug = poll.slug
+
+            if (poll.slug != null && this.slug != poll.slug) {
+                if (isSlugUnique(poll.slug)) {
+                    this.slug = poll.slug
+                } else {
+                    pollRepository.saveAndFlush(this)
+                    throw ResponseStatusException(HttpStatus.BAD_REQUEST, "Slug already exists")
+                }
             }
-            this.currentItem = poll.currentItem
-            webSocketService.sendCurrentItem(this.slug, this.id, this.currentItem)
+
+            if (this.currentItem != null) {
+                webSocketService.sendCurrentItem(this.slug, this.id, this.currentItem)
+                webSocketService.sendItemWithAnswers(this.currentItem!!)
+            }
+
             return pollRepository.saveAndFlush(this).toDtoOut()
         }
     }
 
+    /**
+     * Set the active item from the poll to the next following item.
+     *
+     * @param pollId the id of the poll where the active item is to be continued
+     * @return the next poll item in dto format
+     */
     fun getNextPollItem(pollId: Long): PollItemDtoOut? {
         pollRepository.findById(pollId).orElseThrow {
             ResponseStatusException(HttpStatus.NOT_FOUND, "This poll does not exist")
         }.run {
             pollItems.sortBy { it.position }
-            if (currentItem == null) {
-                this.currentItem = pollItems[0].id
-            } else if (currentItem == pollItems.last().id) {
-                this.currentItem = null
-            } else {
-                val oldItem = pollItems.find { it.id == currentItem }
-                val newItem = pollItems.find { it.position == requireNotNull(oldItem).position + 1 }
-                this.currentItem = requireNotNull(newItem).id
+            when (currentItem) {
+                null -> this.currentItem = pollItems[0].id // set to first item
+                pollItems.last().id -> this.currentItem = null // end reached
+                else -> {
+                    val oldItem = pollItems.find { it.id == currentItem }
+                    val newItem = pollItems.find { it.position == requireNotNull(oldItem).position + 1 }
+                    this.currentItem = requireNotNull(newItem).id
+                }
             }
             webSocketService.sendCurrentItem(this.slug, this.id, this.currentItem)
             pollRepository.saveAndFlush(this)
+
             return if (this.currentItem != null) {
                 pollItemService.getPollItem(this.currentItem!!)
```

Comment:
`this.currentItem` might be `null`. Once we've reached the end of the poll items in the poll, you explicitly set it to null in line 193. So, `this.currentItem!!` is not a good approach here since we might end up calling `getPollItem(null)` in the end...

Reply:
In the line before (line 203) i check that the current item is not null. I only need the "!!" because we use kotlin. In line 205 i return null if the current item is null

Reply:
oh, how could I miss line 203 😅